### PR TITLE
Correct remove_dir_all_where recursion handling / Fix #35

### LINKED ISSUE #35: remove_dir_all_where does not properly handle recursion
It correctly handles the case where the files are left in the directory, but doesn't handle if a sub-directory was not deleted.

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Fixes https://github.com/trussed-dev/littlefs2/issues/35

### CODE REVIEW

File: src/tests.rs [hunk 1]
```diff
@@ -362,6 +363,29 @@ fn test_fancy_open() {
...
+        fs.write(path!("test_file"), b"some data").unwrap();
+        fs.create_dir(path!("test_dir")).unwrap();
+        fs.write(path!("test_dir/test_file"), b"some_inner_data")
+            .unwrap();
+        fs.write(path!("test_dir/test_file2"), b"some_inner_data")
+            .unwrap();
+        fs.remove_dir_all_where(path!(""), &|entry| {
+            entry.path() != path!("test_dir/test_file")
+        })
+        .unwrap();
```
Comment (robin-nitrokey) [inline src/tests.rs:383]: Check that `test_dir/test_file` still exists after `remove_dir_all_where`?

File: src/fs.rs [hunk 2]
```diff
@@ -308,14 +308,14 @@ impl<Storage: driver::Storage> Filesystem<'_, Storage> {
         self.remove_dir_all_where(path, &|_| true).map(|_| ())
     }
 
-    #[cfg(feature = "dir-entry-path")]
-    pub fn remove_dir_all_where<P>(&self, path: &Path, predicate: &P) -> Result<usize>
+    /// Returns number of deleted files + whether the directory was fully deleted or not
+    fn remove_dir_all_where_inner<P>(&self, path: &Path, predicate: &P) -> Result<(usize, bool)>
```
Comment (robin-nitrokey) [inline src/fs.rs:312]: Helper type instead of tuple?

**Review by robin-nitrokey — APPROVED** [review]
LGTM.

File: src/tests.rs (hunk 1 above)
Comment (sosthene-nitrokey) [reply src/tests.rs:383]: Done

**Pull request merged**